ui/pages/confirmations/components/confirm/header/header-info.tsx

```diff
@@ -56,16 +57,35 @@ const HeaderInfo = () => {
 
   const { balance: balanceToUse } = useBalance(fromAddress);
 
+  const isSignatureConfirmation =
+    currentConfirmation &&
+    (currentConfirmation as SignatureRequestType)?.msgParams
+      ?.signatureMethod !== undefined;
+
+  const eventProps = isSignatureConfirmation
+    ? {
+        location: MetaMetricsEventLocation.SignatureConfirmation,
+        signature_type: (currentConfirmation as SignatureRequestType)?.msgParams
+          ?.signatureMethod,
+      }
+    : {
+        location: MetaMetricsEventLocation.Transaction,
+        transaction_type: currentConfirmation?.type,
+      };
+
   function trackAccountModalOpened() {
-    trackEvent({
-      category: MetaMetricsEventCategory.Transactions,
+    const event = {
+      category: isSignatureConfirmation
+        ? MetaMetricsEventCategory.InpageProvider
```

Comment:
`isSignatureConfirmation` is being checked at 2 places, it should be possible to optimise

Reply:
have changed it a bit, not really happy with what it looks like, if you have a better way please let me know